frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx

```diff
@@ -0,0 +1,165 @@
+    React.ComponentProps<typeof TableHeader>,
+    | "onSearchChange"
+    | "onColumnHiddenChange"
+    | "hiddenColumnIds"
+    | "filterButtonComponent"
+    | "columns"
+  >;
+};
+
+const mockUsers = fakeUsers(100);
+
+type Users = Array<User>;
+
+type SearchState = {
+  term?: string;
+  type?: string;
+};
+
+type Data = NonNullable<FromArray<Users>>;
+
+const columns: ColumnsOf<Data> = [
+  {
+    label: "First Name",
+    id: "firstName",
+    accessor: (row: User) => row.firstName,
+    sortColumnName: "firstName",
+  },
+  {
+    label: "Last Name",
+    id: "lastName",
+    accessor: (row: User) => row.lastName,
+    sortColumnName: "lastName",
+  },
+  {
+    label: "Email",
+    id: "email",
+    accessor: (row: User) => row.email,
+    sortColumnName: "email",
+  },
+  {
+    label: "Edit",
+    id: "edit",
+    accessor: () => <a href="/#">Edit</a>,
+  },
+];
+
+export default {
+  component: BasicTable,
+  title: "API Managed Table",
+  subcomponents: {
+    TableHeader,
+    TableFooter,
+  },
+  args: {
+    header: {
+      addBtn: {
+        label: "Add row",
+        path: "/#",
+      },
+    },
+  },
+} as Meta<ApiManageTableProps>;
+
+const allColumnIds = columns.map((c) => c.id);
+
+const Template: Story<ApiManageTableProps> = (args) => {
+  const { header } = args;
+  const [sortingRule, setSortingRule] = React.useState<SortingRule<Data>>();
+  const [hiddenColumnIds, setHiddenColumnIds] = React.useState<IdType<Data>[]>(
+    [],
+  );
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [pageSize, setPageSize] = React.useState(10);
+  const [, setSearchState] = React.useState<SearchState>({
+    type: "",
+    term: "",
+  });
+
+  const paginatedUsers = React.useMemo(() => {
+    const page = currentPage - 1;
+    const start = page * pageSize;
+    const end = start + pageSize;
+    return mockUsers.slice(start, end);
+  }, [pageSize, currentPage]);
+
+  return (
+    <>
+      <h2 id="heading">API Managed Table</h2>
+      <TableHeader
+        columns={columns}
+        hiddenColumnIds={hiddenColumnIds}
+        filterButtonComponent={
+          <Button onClick={() => action("onOpenFilters")()}>Filters</Button>
+        }
+        onSearchChange={(term, type) => {
+          setSearchState({
+            term,
+            type,
+          });
+          action("onSearchChange")({ term, type });
+        }}
+        searchBy={[
+          { label: "First Name", value: "firstName" },
+          { label: "Last Name", value: "lastName" },
+          { label: "Email", value: "email" },
+        ]}
+        onColumnHiddenChange={(event) => {
+          handleColumnHiddenChange(
+            allColumnIds,
+            hiddenColumnIds,
+            setHiddenColumnIds,
+            event,
+          );
+        }}
+        {...header}
+      />
+      <BasicTable
+        labelledBy="heading"
+        columns={columns}
+        data={paginatedUsers}
+        onSortingRuleChange={setSortingRule}
```

Comment:
Could we get an action logged for sort changes, too?

Reply:
Add actions to all events in [add more actions for events](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4542/commits/d45e569b860e36bb1fc44bd1928db0f9bc249fce).